utils/mariadb/files/mysqld.init

```diff
@@ -116,48 +122,39 @@ start() {
 		$LOGGER "Cannot detect privileges table. You might need to run"
 		$LOGGER "'mysql_install_db \"$args\"'"
 		$LOGGER "to initialize the system tables."
+		$LOGGER "Then hand it ower to MariaDB user"
+		$LOGGER "'chown -Rh \"$my_user:$my_group\" \"$datadir\"'"
 		exit 1
 	fi
 
-	# Start daemon
-	if mysqld_status check_alive; then
-		$LOGGER "server is already running"
-	else
-		for i in "$logdir" "$rundir"; do
-			opts="-m 0750"
-			if ! [ -e "$i" ]; then
-				# $rundir needs to be accessible for
-				# clients
-				if [ "$i" = "$rundir" ]; then
-					opts=
-				fi
-				# shellcheck disable=SC2086
-				mkdir -p $opts "$i"
-				[ -d "$i" ] && chown mariadb:mariadb "$i"
+	for i in "$logdir" "$rundir" "$tmpdir" "$datadir"; do
+		opts="-m 0750"
+		if ! [ -e "$i" ]; then
+			# $rundir needs to be accessible for
+			# clients
+			if [ "$i" = "$rundir" ]; then
+				opts=
 			fi
-		done
-		# shellcheck disable=SC2154,SC2086
-		"$MYSQLDSAFE" $options >/dev/null 2>&1 &
-	fi
-}
-
-stop() {
-	timeout="0"
-	while mysqld_status check_alive && [ "$timeout" -lt 60 ]; do
-		mysql_kill -TERM
-		sleep 1
-		timeout="$(($timeout + 1))"
+			# shellcheck disable=SC2086
+			mkdir -p $opts "$i"
+			[ -d "$i" ] && chown -Rh "$my_user:$my_group" "$i"
+		fi
 	done
-	if ! mysqld_status check_dead; then
-		$LOGGER "server is failing to stop"
-		mysql_kill -KILL
-	fi
-}
 
-reload() {
-	if mysqld_status check_alive; then
-		mysql_kill -HUP
-	else
-		$LOGGER "server is not running"
-	fi
+	# Start daemon
+	procd_open_instance
+
+	# shellcheck disable=SC2086
+	procd_set_param command "$MYSQLD" $options
+	procd_set_param respawn "${respawn_threshold:-3600}" "${respawn_timeout:-5}" "${respawn_retry:-5}"
+	# run as user
+	procd_set_param user "$my_user"
+	# forward stderr to logd
+	procd_set_param stderr 1
+	# use HUP to reload
+	procd_set_param reload_signal HUP
```

Comment:
this can be replaced with procd_send_signal I believe.

Reply:
From what I understood from https://openwrt.org/docs/guide-developer/procd-init-scripts it seemed that I can either define my own reload and use `procd_send_signal` in it or I can pass the signal to procd and used the already defined reload function. Did I misunderstood?

Reply:
Hmm good point. I've always seen HUP implemented in a custom reload function.

```
~/d/packages (master)> git grep procd_send_signal | wc -l
16
~/d/packages (master)> git grep reload_signal | wc -l
2
```

The last two are

```
admin/backuppc/files/backuppc.init:    procd_set_param reload_signal 1
net/haproxy/files/haproxy.init: procd_set_param reload_signal USR2
```

Reply:
I don't mind changing that, just wondering if there is some difference I missed in functionality.